BaragonService/src/main/java/com/hubspot/baragon/service/elb/ApplicationLoadBalancer.java

```diff
@@ -29,23 +62,401 @@ public ApplicationLoadBalancer(Optional<ElbConfiguration> configuration,
+   * or which otherwise need to be removed.
+   *
+   * @param targetGroup TargetGroup to check for old agents
+   * @param agents Known agents, to be used as a reference sheet
+   */
+  private void deregisterIfOld(BaragonGroup baragonGroup, TargetGroup targetGroup, Collection<BaragonAgentMetadata> agents) {
+    Collection<TargetDescription> targets = targetsInTargetGroup(targetGroup);
+    Collection<TargetDescription> removeableTargets = listOldTargets(baragonGroup, targets, agents);
+
+    for (TargetDescription removeableTarget : removeableTargets) {
+      try {
+        if (isLastHealthyInstance(removeableTarget, targetGroup)
+            && configuration.isPresent()
+            && !configuration.get().isRemoveLastHealthyEnabled()) {
+          LOG.info("Will not de-register target {} because it is last healthy instance in {}", removeableTarget, targetGroup);
+        } else {
+          elbClient.deregisterTargets(new DeregisterTargetsRequest()
+              .withTargetGroupArn(targetGroup.getTargetGroupArn())
+              .withTargets(removeableTarget));
+          LOG.info("De-registered target {} from target group {}", removeableTarget, targetGroup);
+        }
+      } catch (AmazonClientException acexn) {
+        LOG.error("Could not de-register target {} from target group {} due to error",
+            removeableTarget, targetGroup, acexn);
+        exceptionNotifier.notify(acexn, ImmutableMap
+            .of("targetGroup", targetGroup.getTargetGroupName()));
+      }
+    }
+  }
+
+  /**
+   * When this method completes, the target group, the agents, and the loadBalancers are
+   * all on the same VPC.
+   * The target group, each of the agents, and each of the load balancers should think
+   * that they are on the same VPC, otherwise they won't be able to talk to each other.
+   *
+   *
+   * @param targetGroup Group - and consequently all targets - to check
+   * @param agents Agents to check
+   * @param loadBalancers Load balances to check
+   */
+  private void guaranteeSameVPC(TargetGroup targetGroup,
+                                Collection<BaragonAgentMetadata> agents,
+                                Collection<LoadBalancer> loadBalancers) {
+    String vpcId = targetGroup.getVpcId();
+
+    for (BaragonAgentMetadata agent : agents) {
+      if (agent.getEc2().getVpcId().isPresent()) {
+        if (! agent.getEc2().getVpcId().get().equals(vpcId)) {
+          LOG.warn("Agent {} not on same VPC as its target group {}", agent, targetGroup);
+        }
+      } else {
+        LOG.warn("Agent {} not assigned to a VPC", agent);
+      }
+    }
+
+    for (LoadBalancer loadBalancer : loadBalancers) {
+      if (!vpcId.equals(loadBalancer.getVpcId())) {
+        LOG.warn("Load balancer {} on different VPC from target group {}", loadBalancer, targetGroup);
+      }
+    }
+  }
+
+  private void guaranteeAzEnabled(Collection<BaragonAgentMetadata> agents,
```

Comment:
Read over the docs here again. In the old classic the az/subnet operations were a bit more separated. I think here we can get away with gathering the list of subnets from the agents and feeding that to `guaranteeHasAllSubnets`, rather than doing the az check and than having to check subnets after